packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1116,6 +1145,20 @@ export class ContainerRuntime
 		this.maxOpsSinceLastSummary = this.getMaxOpsSinceLastSummary();
 		this.initialSummarizerDelayMs = this.getInitialSummarizerDelayMs();
 
+		// Enabling the compressor is a one-way operation. Once it's enabled for a document
+		// it is persisted in the container's metadata.
+		const compressorEnabled =
+			metadata?.idCompressorEnabled ??
```

Comment:
This isn't right. For existing documents, you should only look into the metadata and not at options:
```
if (existing) {
  compressorEnabled = metadata?.idCompressorEnabled;
} else {
  // read from feature flags / options.
}

Otherwise, existing documents will enable this feature if the options is set which is not what you want. There is an if (existing) block in the beginning of this constructor, I would put it there.

Reply:
Curious why tests didn't catch this. Do you have a test for this scenario?

Reply:
Good catch, I didn't think about the case of "not letting existing containers enable it" so I didn't have a test and didn't write it correctly at first. Changed to this way and added a test to catch this.